### ISSUE TITLE
train 10 character plate

### ISSUE THREAD

**Issue by BaroDuong16** [ankandrew/fast-plate-ocr#72]
I am training a license plate recognition model with 10-character plates. My dataset includes both 10-character plates and plates with fewer than 10 characters. For plates with fewer than 10 characters, I add ‘_’ padding to reach 10 characters for training. Is this approach good for the model? I have tested it and observed that the model’s predictions are reasonably good

<img width="289" height="97" alt="Image" src="https://github.com/user-attachments/assets/92cdb612-96df-4b72-9cac-ddbb580ce00d" />

**Comment by BaroDuong16** [comment]
I changed max_slots_plate to 10 while still using a pretrained model with max_slots_plate = 9, and I trained on a dataset that includes both 10-character plates and plates with fewer than 10 characters, using the same padding labels as described above. With this setup, the model predicts both 10-character plates and shorter plates quite well. Previously, when I trained the dataset using only 10-character plates and did not include shorter plates, the model produced incorrect predictions for plates with fewer than 10 characters.

and how many epochs should I train for when using a pretrained model. My dataset currently contains approximately 1,100 images

By the way, I really appreciate your work.

**Comment by ankandrew** [comment]
> Is this approach good for the model?

You don't need to include the padding character for those plates with length shorter than `max_slots_plate`, the padding is done automatically. So for example 3rd entry in your screenshot, you can just annotate as `54L19999` (w/o padding char).

> and how many epochs should I train for when using a pretrained model. My dataset currently contains approximately 1,100 images

Depends on other factors such as `learning_rate`, and prob the best way to find it out is experimenting and validating metrics yourself.

> By the way, I really appreciate your work.

I'm glad you find it useful :)
